src/components/RadioButton/RadioButton.vue

```diff
@@ -0,0 +1,190 @@
+<template>
+  <div class="ph-radio-item" :class="[disabled && 'ph-radio-disabled']">
+    <input
+      :id="id"
+      ref="radioInputItem"
+      type="radio"
+      class="ph-radio-input"
+      :name="name"
+      :checked="checked"
+      @click="$emit('click')"
+    />
+    <label
+      :for="id"
+      class="ph-radio-label"
+      :class="icon && 'ph-radio-bar-bottom-brand'"
+    >
+      <div :class="icon && 'ph-radio-label-icon'">
+        <p-icon v-if="icon" :name="icon" type="xl" class="ph-radio-svg-icon" />
+        <span class="ph-radio-label-text">{{ label }}</span>
+      </div>
+    </label>
+
+    <div v-if="!icon" class="ph-radio-circle-container">
+      <div class="ph-radio-circle" />
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue';
+import PIcon from '@/components/Icon';
+
+export default Vue.extend({
+  name: 'InputRadio',
+  components: {
+    PIcon,
+  },
+  props: {
+    name: {
+      type: String as PropType<string>,
+      default: '',
+    },
+    label: {
+      type: String as PropType<string>,
+      default: '',
+    },
+    id: {
+      type: String as PropType<string>,
+      default: '',
+    },
+    icon: {
+      type: String as PropType<string>,
+      default: '',
+    },
+    disabled: {
+      type: Boolean as PropType<boolean>,
+      default: false,
+    },
+    checked: {
+      type: Boolean as PropType<boolean>,
+      default: false,
+    },
+  },
+});
+</script>
+
+<style lang="scss" scoped>
+.ph-radio-item {
+  position: relative;
+}
+.ph-radio-label {
+  display: flex;
+  align-items: center;
+  overflow: hidden;
+  position: relative;
+  padding: var(--radio-button-base-padding);
+  transition-property: background-color, border-color, color, fill, stroke,
+    opacity, box-shadow, transform;
+  font-weight: var(--radio-button-base-font-weight);
+  height: 100%;
+  border-radius: var(--radio-button-base-border-radius);
+  border-width: var(--radio-button-base-border-width);
+  border-style: solid;
+  cursor: pointer;
+  border-color: var(--radio-button-styles-primary-default-border-color);
+  background: var(--radio-button-styles-primary-background);
+  color: var(--radio-button-styles-primary-default-color);
+}
+.ph-radio-disabled .ph-radio-label {
+  cursor: not-allowed;
+  pointer-events: none;
+  opacity: 0.5;
+  color: var(--radio-button-styles-primary-default-color);
+}
+
+.ph-radio-input {
+  position: absolute;
+  left: 0.75rem;
+  top: 50%;
+  bottom: 50%;
+  margin-top: -0.5rem;
+  width: 0;
+  height: 0;
+  opacity: 0;
+  background-color: var(--radio-button-primary-default-border-color);
+  &.focused {
+    outline: none;
+  }
+}
+.ph-radio-input:checked ~ .ph-radio-circle-container {
```

Comment:
**issue:** use nested scss classes

Same here. you can use the nesting functionality of scss to make this cleaner.

```scss
.ph-radio-input {
  // ... basic stuff
  
  &:checked {
    + .ph-radio-label {
    //...
    }
    
    ~ .ph-radio-label {
    //...
    }
  }
}
```